app/src/main/java/com/educacionit/biciya/home/view/MapFragment.kt

```diff
@@ -59,4 +72,40 @@ class MapFragment : Fragment(), OnMapReadyCallback {
         googleMap?.moveCamera(CameraUpdateFactory.newLatLngZoom(position, 16f))
     }
 
+    override fun initPresenter() {
+        presenter = MapPresenterImpl(this@MapFragment, MapModelImpl())
+        presenter.loadStations()
+    }
+
+    override fun showStationOnMap(stations: List<Station>) {
+        googleMap?.let { map ->
+            map.clear() //Limpio el mapa para que no queden estaciones previas marcadas en el mapa, revisar si se agregan mas marcas que NO deban limpiarse.
+
+            val boundsBuilder = LatLngBounds.builder()
+
+            for (station in stations) {
+                val position = LatLng(station.lat, station.lon)
+                map.addMarker(
+                    MarkerOptions()
+                        .position(position)
+                        .title(station.name)
+                )
+                boundsBuilder.include(position)
+            }
+
+            val bounds = boundsBuilder.build()
+            val cameraUpdate = CameraUpdateFactory.newLatLngBounds(bounds, 100)
+            map.animateCamera(cameraUpdate)
+        }
+    }
```

Comment:
Move this code to a another class related to Maps features, to leave the Fragment as clean as possible.